src/glue/jobs/s3_to_json.py

```diff
@@ -16,6 +16,237 @@
+                    except json.JSONDecodeError as e:
+                        # If it's not propertly formatted JSON, then we
+                        # can't read it, and instead store an empty list
+                        logger.warning(f"Problem CustomFields.{field_name}: "
+                                       f"{json_obj['CustomFields'][field_name]}")
+                        logger.warning(str(e))
+                        json_obj["CustomFields"][field_name] = []
+                else:
+                    json_obj["CustomFields"][field_name] = []
+    garmin_transform_types = {
+            "GarminDailySummary": {
+                "TimeOffsetHeartRateSamples": (("OffsetInSeconds", int), ("HeartRate", int))
+            },
+            "GarminHrvSummary": {
+                "HrvValues": (("OffsetInSeconds", int), ("Hrv", int))
+            },
+            "GarminPulseOxSummary": {
+                "TimeOffsetSpo2Values": (("OffsetInSeconds", int), ("Spo2Value", int))
+            },
+            "GarminRespirationSummary": {
+                "TimeOffsetEpochToBreaths": (("OffsetInSeconds", int), ("Breaths", float))
+            },
+            "GarminSleepSummary": {
+                "TimeOffsetSleepSpo2": (("OffsetInSeconds", int), ("Spo2Value", int)),
+                "TimeOffsetSleepRespiration": (("OffsetInSeconds", int), ("Breaths", float))
+            },
+            "GarminStressDetailSummary": {
+                "TimeOffsetStressLevelValues": (("OffsetInSeconds", int), ("StressLevel", int)),
+                "TimeOffsetBodyBatteryValues": (("OffsetInSeconds", int), ("BodyBattery", int))
+            },
+            "GarminThirdPartyDailySummary": {
+                "TimeOffsetHeartRateSamples": (("OffsetInSeconds", int), ("HeartRate", int))
+            },
+            "GarminHealthSnapshotSummary": {
+                "Summaries.EpochSummaries": (("OffsetInSeconds", int), ("Value", float))
+            }
+    }
+    if dataset_identifier in garmin_transform_types:
+        this_data_type = garmin_transform_types[dataset_identifier]
+        for prop in this_data_type:
+            key_name = this_data_type[prop][0][0]
+            key_type = this_data_type[prop][0][1]
+            value_name = this_data_type[prop][1][0]
+            value_type = this_data_type[prop][1][1]
+            property_hierarchy = prop.split(".")
+            if len(property_hierarchy) == 1:
+                prop_name = property_hierarchy[0]
+                if prop_name in json_obj:
+                    array_of_obj = transform_object_to_array_of_objects(
+                            json_obj_to_replace=json_obj[prop_name],
+                            key_name=key_name,
+                            key_type=key_type,
+                            value_name=value_name,
+                            value_type=value_type
+                    )
+                    json_obj[prop_name] = array_of_obj
+            if len(property_hierarchy) == 2:
```

Comment:
Nit: the logic under this if statement sort of resembles the logic above : can they be combined to simplify the code?

Reply:
This was something I did intentionally. It seems pretty difficult to write this in a way that can handle an arbitrary amount of nesting of the JSON property. I would need to write a recursive function. The `len == 1` case would be the base case and you can begin to imagine how the recursive case might look with the extra `for...` block in the `len == 2` case. I think a recursive implementation would be more difficult to understand/debug, so unless we need to handle deeper nesting in the future I think it's better to have it implemented more explicitly -- although a bit redundantly.

Reply:
That makes sense.  Can you add a comment here about that?